File: C++/merge_sort.cpp
#include<iostream>
using namespace std;
void Merge(int a[],int lb,int mid,int ub){
	int i,j,k=lb,b[ub];
	i=lb;
	j=mid+1;
	while(i<=mid && j<=ub){
		if(a[i]<=a[j]){
			b[k]=a[i];
			i++;
		}
		else{
			b[k]=a[j];
			j++;
		}
		k++;
	}
	if(j>=ub)
	while(i<=mid){
		b[k]=a[i];
		i++;
		k++;
	}
	if(i>mid)
	while(j<=ub){
		b[k]=a[j];
		j++;
		k++;
	}
	for(k=lb;k<=ub;k++){
		a[k]=b[k];
	}
}
void MergeSort(int a[],int lb,int ub){
	if(lb<ub){
		int mid=(lb+ub)/2;
		MergeSort(a,lb,mid);
		MergeSort(a,mid+1,ub);
		Merge(a,lb,mid,ub);

}

int main(){
	int n,a[100],i;
	cin>>n;
	for(i=0;i<n;i++){
		cin>>a[i];
	}
	MergeSort(a,0,n-1);
	for(i=0;i<n;i++){
		cout<<a[i]<<" ";
	}

}

File: javascript/mergeSort.js
const merge = (arr, left, mid, right) => {
  let x1 = arr.slice(left, mid+1);
  let x2 = arr.slice(mid+1, right+1);
  let i = 0;
  let j = 0;
  let k = left;
  while (i < x1.length && j < x2.length) {
    if (x1[i] < x2[j]) arr[k++] = x1[i++];
    else arr[k++] = x2[j++];
  }
  while (i < x1.length) {
    arr[k++] = x1[i++];
  }
  while (j < x2.length) {
    arr[k++] = x2[j++];
  }
};

const mergeSort = (arr, left, right) => {
  if (left < right) {
    let mid = Math.floor((right + left) / 2);
    mergeSort(arr, left, mid);
    mergeSort(arr, mid + 1, right);
    merge(arr, left, mid, right);
  }
  return arr;
};

console.log(mergeSort([5,6,7,8,9],0,4));